starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,459 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 >= number2) {
+    return number1;
+  } else {
+    return number2;
+  }
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(words) {
+  var longestWord = "";
+  if (words.length === 0) {
+    return null;
+  }
+  for (var i = 0; i < words.length; i++) {
+    if (words[i].length > longestWord.length) {
+      longestWord = words[i];
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(numbers) {
+  var summed = 0;
+  for (var i = 0; i < numbers.length; i++) {
+    summed += numbers[i];
+  }
+  return summed;
+}
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbersAvg) {
+  var summed = 0;
+  if (numbersAvg.length === 0) {
+    return null;
+  }
+  for (var i = 0; i < numbersAvg.length; i++) {
+    summed += numbersAvg[i];
+  }
+  var av = summed / numbersAvg.length;
+  return av;
+}
+
 // Level 2: Array of strings
```

Comment:
Well done!
Small tip.
I would avoid using var as is getting obsolete and it can bring to many problems in the future when coding in a large scale.
Use` let` for changing variables and `const` for fixed variables instead.